src/common/src/types/datetime.rs

```diff
@@ -364,7 +364,19 @@ impl ToText for Time {
 
 impl ToText for Timestamp {
     fn write<W: std::fmt::Write>(&self, f: &mut W) -> std::fmt::Result {
-        write!(f, "{}", self.0)
+        let (ce, year) = self.0.year_ce();
+        if ce {
+            write!(f, "{}", self.0.date())?;
+        } else {
+            write!(f, "{:04}-{:02}-{:02}", year, self.0.month(), self.0.day())?;
```

Comment:
So will `.date()` return "correct" format for BC? I mean will it return things like `0010-01-01 BC`?

And OTOH, since we are doing a different format from `.date()` here, why not just `write!(f, "{:04}-{:02}-{:02}", year, self.0.month(), self.0.day())?;` for both CE and BC?

Reply:
> So will .date() return "correct" format for BC? I mean will it return things like 0010-01-01 BC?

No, it will return `-0010-01-01`

> for both CE and BC?

Updated.